jdaviz/configs/default/plugins/subset_tools/subset_tools.py

```diff
@@ -165,12 +165,110 @@ def __init__(self, *args, **kwargs):
 
     @property
     def user_api(self):
-        expose = ['subset', 'combination_mode',
-                  'recenter_dataset', 'recenter',
-                  'get_center', 'set_center',
-                  'import_region']
+        expose = ['subset', 'combination_mode', 'recenter_dataset', 'recenter',
+                  'get_center', 'set_center', 'import_region', 'get_regions']
         return PluginUserApi(self, expose)
 
+    def get_regions(self, region_type=None, list_of_subset_labels=None,
```

Comment:
you said that using existing code for this is out-of-scope and it has to have its own dedicated logic for now?

Reply:
yep. it would be nice to use get_subsets rather than repeating this logic. the current implementation will also cause issues for subset renaming, since it is looking for layers named 'Subset x'. That ticket for replacing is at the top of the back log. It was out of scope here because there will be some changes with how composite subsets are returned.